src/include/msvc.h

```diff
@@ -19,8 +19,6 @@
 /*
  * Add MSVC-specific attributes and pragmas to types and function declarations.
  */
-#define	WT_COMPILER_TYPE_ALIGN(x)	__declspec(align(x))
```

Comment:
A little more digging and that function is never call, it simply has a bunch of static asserts in it. It is included as part of wt_internal.h, so should be compiled on Windows.